pkg/neonvm/ipam/ipam.go

```diff
@@ -49,30 +50,38 @@ type Temporary interface {
 
 type IPAM struct {
 	Client
-	Config IPAMConfig
+	Config  IPAMConfig
+	metrics *IPAMMetrics
 
 	mu                 sync.Mutex
 	concurrencyLimiter *semaphore.Weighted
 }
 
+type IPAMParams struct {
+	NadName          string
+	NadNamespace     string
+	ConcurrencyLimit int
+	MetricsReg       prometheus.Registerer
+}
+
 func (i *IPAM) AcquireIP(ctx context.Context, vmName types.NamespacedName) (net.IPNet, error) {
-	ip, err := i.runIPAM(ctx, makeAcquireAction(ctx, vmName))
+	ip, err := i.runIPAMWithMetrics(ctx, makeAcquireAction(ctx, vmName), IPAMAcquire)
 	if err != nil {
 		return net.IPNet{}, fmt.Errorf("failed to acquire IP: %w", err)
 	}
 	return ip, nil
 }
 
 func (i *IPAM) ReleaseIP(ctx context.Context, vmName types.NamespacedName) (net.IPNet, error) {
-	ip, err := i.runIPAM(ctx, makeReleaseAction(ctx, vmName))
+	ip, err := i.runIPAMWithMetrics(ctx, makeReleaseAction(ctx, vmName), IPAMRelease)
 	if err != nil {
 		return net.IPNet{}, fmt.Errorf("failed to release IP: %w", err)
 	}
 	return ip, nil
 }
 
 // New returns a new IPAM object with ipam config and k8s/crd clients
-func New(nadName string, nadNamespace string, concurrencyLimit int) (*IPAM, error) {
+func New(params *IPAMParams) (*IPAM, error) {
```

Comment:
question(non-blocking): Does params need to be a pointer?

Reply:
No, but why is one option is better than the other?

Reply:
Just thinking it's worth avoiding pointers if they don't provide any value, and it looked like that might be the case here? If not, feel free to leave as is

Reply:
I've already changed it, I'll follow-up later on that topic.